src/constructs/core/ssm.ts

```diff
@@ -0,0 +1,90 @@
+import { readFileSync } from "fs";
+import { join } from "path";
+import type { IGrantable, IPrincipal } from "@aws-cdk/aws-iam";
+import { Policy, PolicyStatement } from "@aws-cdk/aws-iam";
+import { Code, Runtime, SingletonFunction } from "@aws-cdk/aws-lambda";
+import type { Reference } from "@aws-cdk/core";
+import { Construct, CustomResource, Duration } from "@aws-cdk/core";
+import { AwsCustomResourcePolicy } from "@aws-cdk/custom-resources";
+import type { CustomResourceGetParameterProps } from "./custom-resources/interfaces";
+import type { GuStack } from "./stack";
+
+export interface GuSSMParameterProps {
+  secure?: boolean;
+  /*
+   * Assumes the path of `/${STAGE}/${STACK}/${APP}/${param}`
+   * */
+  defaultPath: boolean;
+}
+
+const stripped = (str: string) => str.replace(/[-/]/g, "");
+
+export class GuSSMParameter extends Construct implements IGrantable {
+  private readonly customResource: CustomResource;
+  readonly grantPrincipal: IPrincipal;
+
+  // eslint-disable-next-line custom-rules/valid-constructors -- I think stating an ID would be overkill for this, but happy to discuss
+  constructor(scope: GuStack, param: string, props?: GuSSMParameterProps) {
+    // TODO: Establish if this is an accepted, safe way of creating IDs
+    const id = (id: string) =>
+      param.toUpperCase().includes("TOKEN") ? `${id}-token-${Date.now()}` : `${id}-${stripped(param)}`;
+
+    super(scope, id("GuSSMParameter"));
+
+    const provider = new SingletonFunction(scope, id("Provider"), {
+      code: Code.fromInline(readFileSync(join(__dirname, "/custom-resources/runtime/lambda.js")).toString()),
+      // runtime: new Runtime("nodejs14.x", RuntimeFamily.NODEJS, { supportsInlineCode: true }), -- we can use Node14 once we bump the version of @aws-cdk to v1.94 https://github.com/aws/aws-cdk/releases/tag/v1.94.0
```

Comment:
I think you're [in luck](https://github.com/guardian/cdk/pull/327)!

Reply:
Yay! Updated to use Node 14.